### PR TITLE
Stop motor test with Any key

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #2918

- [x] Add `ANY` key handler to stop motor test.
- [x] Add `Stop Motors` button. 
- [x] Only poll MSP status updates when testing is enabled.
- [x] Updates MSP promises to `async await`.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
It does not seem a bad idea to use the space bar to stop de motos, but it will not fix the issue. The major part of users will ignore this. If we want a really emergency button:
- Something visible needs to be in the page. The major part of users will not remember any text. This will not work for Android, for example, the keyboard is not shown.
- Maybe capture ANY key and not only space? Being an emergency, if the motors are on, any key can stop the motors. Only ignore the ones that move the sliders.

**Comment by haslinghuis** [comment]
@McGiverGim Now any key stops testing. Didn't want to update text to avoid new translation cycle.


press-any-key

**Comment by Asizon** [comment]
@haslinghuis i think that the correct aproach here is to add a visible button in motor tab. This also will work in non keyboard devices.

**Comment by McGiverGim** [comment]
I think too that a visible button is the correct final solution, but at least stopping with any key press is a step forward in security.

If this is going to be merged in Configurator 10.8, I think we need to change the text explaining this now. 

The translators will need to translate it again, but the worst that can happen is that this remain in English if the translator don't do it. In the other side, the worst that can happen if we don't modify the text is that nobody knows about a safety feature.

**Comment by haslinghuis** [comment]
Added a stop motor button already translated.

**Review by McGiverGim — APPROVED** [review]
I think the "stop motors" button must be a red, big button, in the middle of the screen. Is an emergency button, not an ordinary one.
But I think is difficult to add more things to this screen, so I approved it. Better to have it than not to have.

**Comment by KarateBrot** [comment]
> @McGiverGim 
Maybe capture ANY key and not only space? Being an emergency, if the motors are on, any key can stop the motors. Only ignore the ones that move the sliders.


Not ANY key, because we can increase/decrease motor output with the arrow keys, for example.

**Review by limonspb — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
